[REF] Planning: create new field service page in planning (#17905)
odoo/documentation: content/applications/services/planning/field_service.rst

   requesting location permission is usually displayed the first time a user signs in to an
   intervention after the feature is enabled. If the pop-up does not appear, users must enable
   location access manually in the browser's site settings.

.. _planning/field-service/worksheets:

Worksheets
==========

Worksheets are used to document and report on work performed during on-site interventions. They can
include various types of content, such as instructions, checklists, text fields, etc.

.. _planning/field-service/worksheet-configuration:

Configuration
-------------

To use worksheets for field service interventions, go to :menuselection:`Planning --> Configuration
--> Settings`, enable :guilabel:`Worksheets` in the :guilabel:`Field Service` section, and click
:guilabel:`Save`.

Then, create worksheet templates:

#. Go to :menuselection:`Planning --> Configuration --> Worksheet Templates`.
#. Click :guilabel:`New`.
#. Enter a :guilabel:`Name`.
#. Under :guilabel:`Worksheet properties`, click :icon:`fa-plus` :guilabel:`Add Property`, then
   :ref:`configure the property field <essentials/property-fields/add>`.

> #. Repeat for each additional property field.

Reviewer: Maybe nitpicking

```suggestion
#. Repeat the previous step for each additional property field.
```

#. Go to the `Access tokens <https://account.mapbox.com/access-tokens/>`_ page on MapBox and copy
   the token.
#. In Odoo, open the Settings app and scroll down to the :guilabel:`Integrations` section.
#. Paste the MapBox access token in the :guilabel:`Token` field under :guilabel:`Map Routes`, then
   click :guilabel:`Save`.

Once enabled, the technician's map displays a complete itinerary for the day, which can be
opened on the Google Maps website or app by clicking :guilabel:`Maps`. Google Maps uses the current
location as the starting point.

.. _planning/field-service/invoicing:

Invoicing
=========

To invoice customers for services performed and :ref:`materials <planning/field-service/products>`
used during interventions, go to :menuselection:`Planning --> Configuration --> Settings` and
enable :guilabel:`Billing` in the :guilabel:`Field Service` section.

  .. note::
     - Enabling the :guilabel:`Billing` feature automatically installs the :doc:`Sales app
       </applications/sales/sales>`.
     - The :ref:`Products <planning/field-service/products>` and :ref:`Customer Report
       <planning/field-service/customer-report>` features are enabled by default when

>        :guilabel:`Billing` is activated.

Reviewer: Sorry, I don't remember what we discussed yesterday, but it seems the Customer Report isn't automatically enabled when Billing is enabled. It is just automatically enabled when enabling the FIeld Service feature. Only the Products feature depends on the Billing feature.

   - :doc:`Odoo Sales documentation </applications/sales/sales>`

Time-based invoicing
--------------------

To invoice the actual time spent on an intervention, the :doc:`Project <../project>` and
:doc:`Timesheets <../timesheets>` apps must be installed, and the following must be defined on the
:ref:`shift form <planning/create-shift>`:

- :guilabel:`Resources`: Assign at least one employee to the shift.
- :guilabel:`Project`: The project linked to the shift must be configured as :guilabel:`Billable`,
  and :guilabel:`Timesheets` must be enabled in the :ref:`project's settings
  <project_management/configuration>`.

The sales order's service is then invoiced based on the shift's :guilabel:`Allocated Time`, which is
updated upon shift :ref:`start and completion <planning/field-service/record>`.

.. _planning/field-service/employee-service:

If an employee's work is billed at a specific rate or with extra fees, define a dedicated product
on the employee's form. To do so, go to :menuselection:`Planning --> Configuration --> Employees`,
select the relevant employee, go to the :guilabel:`Settings` tab, and select the relevant

> :guilabel:`Service to Bill` in the :guilabel:`Planning` section.

Reviewer: I think you rephrased it really well. It is clearer now :D

add more.

.. tip::
   - To locate a product more easily in the catalog, use the search bar and filter by
     :guilabel:`Product Category` or :ref:`Attributes <products/variants/attributes>`.
   - To create or edit products from the Planning app, go to :menuselection:`Planning -->
     Configuration --> Products`.

.. note::
   - Adding products from a shift requires the :guilabel:`User: All Documents` (or
     :guilabel:`Administrator`) :guilabel:`Sales` :doc:`access right
     </applications/general/users/access_rights>`.
   - Any products added to the shift are added to the related sales order. If no sales order is
     linked to the shift, a new one is created automatically. To access it, click the
     :guilabel:`Sales Order` smart button on the shift form.
   - Products added to shifts marked as :guilabel:`Under Warranty` are not invoiced; their price is
     automatically set to `0` on the sales order.

.. _planning/field-service/quotations:

Quotations
----------

Quotations can be created directly from field service shifts, for example, when additional work or
materials are identified during an intervention. To activate this feature, go to

> :menuselection:`Planning --> Configuration --> Settings`, ensure the :guilabel:`Billing` setting is
> enabled in the :guilabel:`Field Service` section, enable :guilabel:`Quotations`, and click
> :guilabel:`Save`.

Reviewer: propably nitpicking

```suggestion
:menuselection:`Planning --> Configuration --> Settings`, ensure the :guilabel:`Billing` setting is
enabled in the :guilabel:`Field Service` section, then enable :guilabel:`Quotations` and click
:guilabel:`Save`.
```

To create a quotation from a shift, click the :guilabel:`Quotations` smart button on the shift form,
then click :guilabel:`New` and :ref:`fill in the fields <sales/create-quotation>`.

.. _planning/field-service/customer-report:

Customer report
===============

To automatically generate a customer report upon shift completion, go to :menuselection:`Planning
--> Configuration --> Settings`, ensure the :guilabel:`Billing` setting is enabled in the

> :guilabel:`Field Service` section, enable :guilabel:`Customer Report`, and click :guilabel:`Save`.

Reviewer: See my comment in line 205.
The Customer Report feature does not seem to depend on the Billing feature.

<img width="1017" height="233" alt="Image" src="https://github.com/user-attachments/assets/bf7e85b3-9ead-403d-b6c5-544bc83dd793" />

The report details the materials and services provided, the time spent, and the completed worksheet,
if applicable. It is generated automatically at the end of the intervention if the intervention
is :ref:`billed <planning/field-service/invoicing>` or a :ref:`worksheet
<planning/field-service/worksheets>` has been completed.

Click the :guilabel:`Customer Preview` smart button on the shift form to access the report, then
click :guilabel:`Sign Report` to collect the customer's :ref:`signature
<sign/sign-document/initials-signature>` and send the report to the customer.

.. tip::
   To exclude prices from the customer report for :ref:`billed <planning/field-service/invoicing>`
   interventions, enable :guilabel:`Products`, then :guilabel:`Hide prices on customer report` in

>    the Planning settings.

Reviewer: We could add the reference to the [Products](planning/field-service/products) section to make sure people know how to enable.

=============
Field service
=============

> The field service feature allows on-site interventions to be scheduled directly in the Planning

Reviewer: I know "Field Service" is a feature you can enable, but I wouldn't call it a "feature" here; it's more of a module or even an App (it is still under the "Apps" filter). What do you think?

<img width="726" height="219" alt="Image" src="https://github.com/user-attachments/assets/0125a9ac-a1bc-4614-84e5-33366b7aa55b" />

app. Work performed can be recorded using custom :ref:`worksheets
<planning/field-service/worksheets>`, while :ref:`materials used
<planning/field-service/products>` and time spent can be tracked on the :ref:`shift
<planning/field-service/record>` and billed to the customer. A customer report can also be
generated at the end of the intervention for signature, and :ref:`itineraries

> <planning/field-service/itinerary>` can be planned for field employees.

Reviewer: ```suggestion
<planning/field-service/record>` and billed to the customer. At the end of the intervention, a customer report can be generated for signature, and :ref:`itineraries
<planning/field-service/itinerary>` can be planned for field employees.
```

To activate the feature, go to :menuselection:`Planning --> Configuration --> Settings`, enable
:guilabel:`Field Service`, and click :guilabel:`Save`.

.. _planning/field-service/shifts:

Plan an intervention
====================

Interventions are planned as :ref:`shifts <planning/shifts>` and can be created:

- directly from the Planning app, in the same way as standard :ref:`shifts <planning/create-shift>`;
- :ref:`from sales orders <planning/shifts-sales-orders>`;
- :ref:`from helpdesk tickets <helpdesk/after-sales/field-service>`.

Field service-specific information is added manually or automatically, depending on the use case,

> using the following fields on the shift form:

Reviewer: ```suggestion
Depending on the use case, field service-relevant information is added either manually or automatically using the following fields on the shift form:
```

Maybe better: "relevant," since these fields are not all Field service-specific; e.g., the "Sales Order" field is still displayed even when the Field Service setting is disabled.

- :guilabel:`Customer`: Select or create the relevant customer.
- :guilabel:`Sales Order`: If the intervention must be :ref:`billed

>   <planning/field-service/billing>`, select the relevant sales order. Only the sales orders linked

Reviewer: So it is not possible to bill an intervention without a sales order?

Reviewer 2: Technically, from what I tested, a sales order must be linked to the shift for the shift to be billed (see explanation below from the Invoicing section). BUT, you don't necessarily need to add the sales order at this stage so we should probably change this sentence, you're right. Any ideas here?

<img width="897" height="164" alt="Image" src="https://github.com/user-attachments/assets/30b902db-3c5b-42a9-8eaf-2c854ad5f3f2" />

Reviewer 2: Maybe just removing the first part ("if the intervention must be billed")? I think I would still want the link to the invoicing section, or is that not necessary you think?

  to the specified customer and containing service products are available in the dropdown menu. If
  the shift was :ref:`created from a sales order <planning/shifts-sales-orders>`, this field is
  filled automatically.
- :guilabel:`Worksheet`: If :ref:`worksheets <planning/field-service/worksheets>` are enabled, the
  default one is automatically selected in this field. Choose a different one from the dropdown menu
  if needed.
- :guilabel:`Under warranty`: Enable the checkbox to prevent any material or service :ref:`added to
  the sales order from the shift <planning/field-service/products>` from being invoiced, by
  automatically setting its price to zero. The price of any material or service added directly on

>   the sales order remains unchanged.

Reviewer: ```suggestion
- :guilabel:`Under warranty`: Enable the checkbox to automatically set the price of any materials or services :ref:`added to the sales order while working on the shift <planning/field-service/products>` to zero, preventing them from being invoiced. The price of any material or service added directly to
  the sales order form remains unchanged.
```

.. tip::
   Ensure the customer has a valid address if :ref:`itinerary planning
   <planning/field-service/itinerary>` is required.

> Once the shift has been created, :ref:`publish it <planning/shifts-publish>`.

Reviewer: I would move this sentence above the tip. It seems a bit lost here.

> .. seealso::

Reviewer: Not sure if you need this reference here as you have already referenced this link in line 23.

   :ref:`planning/create-shift`

.. _planning/field-service/record:

Sign in and work on an intervention
===================================

.. note::
   In order to start working on an intervention:

   - the shift must be :ref:`published <planning/shifts-publish>`

>    - a :guilabel:`Customer` must be :ref:`defined for the shift <planning/field-service/shifts>`

Reviewer: ```suggestion
   - The shift must be :ref:`published <planning/shifts-publish>`
   - A :guilabel:`Customer` must be :ref:`defined for the shift <planning/field-service/shifts>`
```

To sign in and start working on an intervention, follow these steps:

> #. Go to :menuselection:`Planning --> Planning --> My Planning`, :menuselection:`Planning -->

Reviewer: Maybe I am missing something, but I thought that you can only sign in for shifts you have been assigned to, don't you? I think you can only sign in for other people's shifts if you have specific access rights...
That's why I wouldn't mention the "All Planning" path, as you cannot see this menu item as a user with extended access rights. You can only see it as an employee with limited access rights, but then you cannot sign in for shifts you have not been assigned to. And since you can only sign in for your own shifts, I wouldn't mention it.
Not sure if you understand what I mean 💩

Reviewer 2: Yes you're 100% right on this, thank you!

   Planning --> All Planning`, or access any :ref:`schedule <planning/shifts>`.
#. Click the relevant intervention in the calendar or schedule, then click :guilabel:`Sign In`.
#. Add the required items to the shift, such as :guilabel:`Photos`, :ref:`Products
   <planning/field-service/products>`, :ref:`Worksheet <planning/field-service/worksheets>` details,

>    and internal notes.

Reviewer: ```suggestion
#. Use the buttons above the form to add the required items to the shift, such as :guilabel:`Photos` or :ref:`Products
   <planning/field-service/products>`.
#. Use the :guilabel:`Note` and :guilabel:`Worksheet` tabs to add :ref:`Worksheet <planning/field-service/worksheets>` details and internal notes.
```
I spent some time looking for the Photos and Products buttons because I couldn't find them in the form. That's why I made this suggestion.

> #. Click :guilabel:`Complete`, then :guilabel:`Complete` to confirm and generate the

Reviewer: I just had to click "Complete" once. Am I missing somthing?

   :ref:`customer report <planning/field-service/customer-report>`, if applicable.

.. note::
   The intervention's start and end date and time are updated upon signing in to and completing it,
   respectively, and the :guilabel:`Allocated Time` is adjusted accordingly.

.. tip::

>    To add items to ongoing shifts at a later point, :ref:`edit the shift <planning/shifts-edit>`.

Reviewer: ```suggestion
.. tip::
   To add items or information to shifts at a later point, :ref:`edit the shift <planning/shifts-edit>`.
```
As you might also add a note or photos? And it seems that you can even add information once it is completed.

Reviewer 2: I suppose I added *ongoing* because the shift must've been signed into for the buttons to be available and I wanted to make that clear without sounding too weird. But you're right that this leaves out the fact that you could also do this for "completed" shifts ...

.. _planning/field-service/geolocation:

Geolocation
-----------

To track the technician's location at the start and end of an intervention, go to
:menuselection:`Planning --> Configuration --> Settings` and enable :guilabel:`Geolocation` in the

> :guilabel:`Field service` section.

Reviewer: ```suggestion
:guilabel:`Field Service` section.
```

Once enabled, users are geolocalized upon clicking :guilabel:`Sign In` and :guilabel:`Complete`
on a shift; the GPS coordinates are logged in the shift's chatter. Click :guilabel:`View on Map`
to view the location in Google Maps.

.. note::
   Geolocation required the browser to be granted access to the user's location. A popup
   requesting location permission is usually displayed the first time a user signs in to an

>    intervention after the feature is enabled. If the popup does not appear, location access has

Reviewer: ```suggestion
   Geolocation required the browser to be granted access to the user's location. A pop-up
   requesting location permission is usually displayed the first time a user signs in to an
   intervention after the feature is enabled. If the pop-up does not appear, location access has
```

   to be enabled manually in the browser's site settings.

.. _planning/field-service/billing:

Invoicing
=========

To invoice customers for services performed and :ref:`materials <planning/field-service/products>`
used during interventions, go to :menuselection:`Planning --> Configuration --> Settings` and

> enable :guilabel:`Billing` in the :guilabel:`Field service` section.

Reviewer: ```suggestion
enable :guilabel:`Billing` in the :guilabel:`Field Service` section.
```

  .. note::
     - Enabling the :guilabel:`Billing` feature automatically installs the :doc:`Sales app
       </applications/sales/sales>`.
     - The :ref:`Products <planning/field-service/products>` and :ref:`Customer Report
       <planning/field-service/customer-report>` features are enabled by default when
       :guilabel:`Billing` is activated.

Once the feature is enabled, an intervention is billed whenever it is linked to a sales order. This
occurs when:

- the intervention was :ref:`planned from a sales order <planning/shifts-sales-orders>`;
- a sales order was :ref:`selected on the shift <planning/field-service/shifts>`;
- :ref:`products were added <planning/field-service/products>` from the shift form, provided the
  shift is not marked as :guilabel:`Under Warranty`.

> - a :guilabel:`Service to Bill` is defined on the form of the employee(s) assigned as the shift's

Reviewer: Where can I find this setting on the employee form? I am not able to find it. Is this something that should be mentioned in the general Planning doc?

EDIT: Found the information in lines 142-146. I would create an anchor in line 141 and reference it here.

>   :guilabel:`Resources`

Reviewer: ```suggestion
- The intervention was :ref:`planned from a sales order <planning/shifts-sales-orders>`;
- A sales order was :ref:`selected on the shift <planning/field-service/shifts>`;
- :ref:`Products were added <planning/field-service/products>` from the shift form, provided the
  shift is not marked as :guilabel:`Under Warranty`;
- A :guilabel:`Service to Bill` is defined on the form of the employee(s) assigned as the shift's
  :guilabel:`Resources`.
```

Time-based invoicing
--------------------

To invoice the actual time spent on an intervention, the :doc:`Project <../project>` and
:doc:`Timesheets <../timesheets>` apps must be installed, and the following must be defined on the
:ref:`shift form <planning/create-shift>`:

> - :guilabel:`Resources`

Reviewer: Maybe add some information what is meant here
```suggestion
- :guilabel:`Resources`: Assign at least one employee to the shift.
```

> - :guilabel:`Project`: the project linked to the shift must be configured as :guilabel:`Billable`

Reviewer: ```suggestion
- :guilabel:`Project`: The project linked to the shift must be configured as :guilabel:`Billable`,
```

  and :guilabel:`Timesheets` must be enabled in the :ref:`project's settings
  <project_management/configuration>`

The sales order's service is then invoiced based on the shift's :guilabel:`Allocated Time`, which is

> updated upon shift start and completion.

Reviewer: Maybe add a reference to the Sign in and work on an intervention section?

If an employee's time must be invoiced differently, for example, to reflect a specific rate or
service type, a dedicated product can be defined on the employee's form. To do so, go to

> :menuselection:`Planning --> Configuration --> Employee`, select the relevant employee, go to the

Reviewer: ```suggestion
:menuselection:`Planning --> Configuration --> Employees`, select the relevant employee, go to the
```

:guilabel:`Settings` tab, and select the relevant :guilabel:`Service to Bill` in the
:guilabel:`Planning` section.

The service defined for the employee is added to the sales order linked to the intervention, if
applicable. Otherwise, a sales order is created with the employee's service when the shift is

> marked as completed, and invoiced based on the time spent on the intervention.

Reviewer: I am not sure if I get this. So, whenever an employee spends time on an intervention for any customer, is this service billed? Sorry might be a dumb question, but I just don't get it 🤔

Reviewer 2: Yes, if a service to bill is configured for the employee, the time spent on the shift is billed based on the related service (product) configuration.

Let's discuss this maybe, I think I need to rephrase.

.. _planning/field-service/products:

> Products

Reviewer: I know the Products feature is only displayed when Billing is enabled, but I would move this section up.  See my general comment

--------

Materials used during interventions can be recorded directly from field service shifts and billed to
customers.

To add products to a shift, ensure the :ref:`Billing <planning/field-service/billing>` and
:guilabel:`Products` features are enabled in the Planning settings. Then, follow these steps:

#. :ref:`Sign in to the shift <planning/field-service/record>` or :ref:`edit <planning/shifts-edit>`
   an ongoing shift.
#. Click the :guilabel:`0 Products` button.
#. Click :icon:`fa-shopping-cart` :guilabel:`Add` on a product card.
#. Adjust the product quantity, if needed, using the :guilabel:`-` and :guilabel:`+` buttons.
#. Click :guilabel:`Back` to return to the shift.

The :guilabel:`Products` button displays the number of products added to the shift. Click it to
add more.

.. tip::
   - To locate a product more easily in the catalog, use the search bar and filter by

>      :guilabel:`Product Category` or :guilabel:`Attributes`.

Reviewer: I would add a reference here

```suggestion
     :guilabel:`Product Category` or :ref:`Attributes <products/variants/attributes>`.
```

   - To create or edit products from the Planning app, go to :menuselection:`Planning -->
     Configuration --> Products`.

.. note::
   - Adding products from a shift requires the :guilabel:`User: All Documents` (or
     :guilabel:`Administrator`) :guilabel:`Sales` :doc:`access right
     </applications/general/users/access_rights>`.
   - Any products added to the shift are added to the related sales order. If no sales order is
     linked to the shift, a new one is created automatically. To access it, click the

>      :guilabel:` Sales Order` smart button on the shift form.

Reviewer: ```suggestion
     :guilabel:`Sales Order` smart button on the shift form.
```

   - Products added to shifts marked as :guilabel:`Under Warranty` are not invoiced; their price is

>      automatically set to 0 on the sales order.

Reviewer: ```suggestion
     automatically set to `0.00` on the sales order.
```

.. seealso::
   :ref:`Invoice a sales order <accounting/inv-process/so>`

>    :doc:`Odoo Sales documentation </applications/sales/sales>`

Reviewer: ```suggestion
.. seealso::
   - :ref:`Invoice a sales order <accounting/inv-process/so>`
   - :doc:`Odoo Sales documentation </applications/sales/sales>`
```

.. _planning/field-service/quotations:

> Quotations

Reviewer: I would move this section closer to "Sign in and work on an intervention", as this is something you might need while working on the intervention. Having it under Invoicing might be misleading (especially since it is not automatically enabled when "Billing" is enabled). What do you think?
See my general comment as well

----------

Quotations can be created directly from field service shifts, for example, when additional work or
materials are identified during an intervention. To activate this feature, go to
:menuselection:`Planning --> Configuration --> Settings` and enable :guilabel:`Quotations` in the

> :guilabel:`Field service` section.

Reviewer: ```suggestion
:guilabel:`Field Service` section.
```

To create a quotation from a shift, click the :guilabel:`Quotations` smart button on the shift form,
then click :guilabel:`New` and :ref:`fill in the fields <sales/create-quotation>`.

.. _planning/field-service/customer-report:

Customer report
---------------

To automatically generate a customer report upon shift completion, the :ref:`Billing
<planning/field-service/billing>` and :guilabel:`Customer Report` features must be enabled in
the Planning settings.

.. tip::
   Enable :guilabel:`Hide prices on customer report` under :guilabel:`Products` in the Planning

>    settings to generate a report without product prices.

Reviewer: We might need to mention that you need to enable the Products setting to see the Hide prices option.

The report includes details of the products used, time spent, and worksheet completed. It is
generated automatically at the end of the intervention if at least one of the following was

> recorded on the shift:

Reviewer: ```suggestion
The report includes details on the products used, the time spent, and the worksheet completed. It is
generated automatically at the end of the intervention if at least one of the following was
recorded on the shift:
```

- :ref:`products <planning/field-service/products>`
- a :ref:`worksheet <planning/field-service/worksheets>`
- a :doc:`timesheet <../timesheets>`

> 

Reviewer: ```suggestion
- :ref:`products <planning/field-service/products>`;
- a :ref:`worksheet <planning/field-service/worksheets>`;
- a :doc:`timesheet <../timesheets>`.

```

I don't know if this is relevant, but we can also add a tip that you can customize the text usinf the website editor.

Reviewer 2: Which text exactly? or do you mean customize the report itself?

Click the :guilabel:`Customer Preview` smart button on the shift form to access the report, then
click :guilabel:`Sign Report` to collect the customer's signature and send the report to the

> customer.

Reviewer: We might add a reference to [this](sign/sign-document/initials-signature) because the Sign field service report pop-up opens when clicking Sign Report

.. _planning/field-service/worksheets:

> Worksheets

Reviewer: I am considering moving the Worksheets sections above Invoicing, as they are relevant before invoicing and more closely related to the "work on an intervention" section. What do you think?
See my general comment as well

==========

Worksheets are used to document and report on work performed during on-site interventions. They can
include various types of content, such as instructions, checklists, text fields, etc.

.. _planning/field-service/worksheet-configuration:

Configuration
-------------

To use worksheets for field service interventions, go to :menuselection:`Planning --> Configuration

> --> Settings`, enable :guilabel:`Worksheets` in the :guilabel:`Field service` section, and click

Reviewer: ```suggestion
--> Settings`, enable :guilabel:`Worksheets` in the :guilabel:`Field Service` section, and click
```

:guilabel:`Save`.

Then, create worksheet templates:

#. Go to :menuselection:`Planning --> Configuration --> Worksheet Templates`.
#. Click :guilabel:`New`.
#. Enter a :guilabel:`Name`.

> #. Click :guilabel:`Add Property`, then :ref:`configure the property field

Reviewer: ```suggestion
#. Under :guilabel:`Worksheet properties`, click :icon:`fa-plus` :guilabel:`Add Property`, then :ref:`configure the property field
```

> #. Click :guilabel:`Add Property`, then :ref:`configure the property field

Reviewer: In a multi-company environment, you can select a Company as well. We might mention that as well.

Reviewer 2: I usually ignore multiwebsite (except in website + ecommerce) + multicompany settings/fields in docs (unless it is relevant and something is really specific to that feature + app), hence why I left it out 😉

   <essentials/property-fields/add>`.
#. Repeat for each additional property field.

.. seealso::
   :ref:`essentials/property-fields/edit`

> 

Reviewer: I don't think this is important here, as you refer to the same page in line 253, and this "Edit the property field" section just follows the previously referenced section. And this depends on whether you take my comment on the Property Fields file into consideration or not 😄

Reviewer 2: I did, so yes I removed the seealso 😉

.. note::

>    By default, the first worksheet in the list view is applied to field service shifts, unless

Reviewer: ```suggestion
   By default, the first worksheet in the list view is applied to field service shifts unless
```

   one is :ref:`manually selected <planning/field-service/worksheet-shift>`. To reorder
   worksheet templates, use the :icon:`oi-draggable` (:guilabel:`drag handle`) icon.

.. tip::
   - Worksheet templates can also be :ref:`created and/or edited from the shift form
     <planning/field-service/worksheet-shift>`.
   - A default worksheet can also be assigned to a specific product. To do so, configure the product
     to automatically :ref:`create a shift <planning/shifts-sales-orders>` from sales orders that

>      contain it, then select the relevant :guilabel:`Worksheet` in the :guilabel:`General

Reviewer: ```suggestion
     contain it, then select the relevant :guilabel:`Worksheet` :icon:`fa-building-o` in the :guilabel:`General
```

     Information` tab of the product form.

.. _planning/field-service/worksheet-shift:

Worksheets in shifts
--------------------

The default :ref:`worksheet template <planning/field-service/worksheet-configuration>` is
automatically selected in the :guilabel:`Worksheet` field on the shift form upon :ref:`creation

> <planning/field-service/shifts>`. Select a different template from the dropdown menu if needed.

Reviewer: We could mention that you can also create a new one from here.

.. tip::
   To edit a worksheet template from the shift, hover over the :guilabel:`Worksheet` dropdown
   menu, click the :icon:`fa-external-link` (:guilabel:`internal link`) icon, then :ref:`edit the
   template <essentials/property-fields/add>`. Changes apply to **all** shifts using that template.

The worksheet can then be completed in the :guilabel:`Worksheet` tab on the shift form :ref:`during
the intervention <planning/field-service/record>`.

.. _planning/field-service/map:

> Map

Reviewer: Same as for Worksheets (see comment above), I would move it above Invoicing

===

Technicians can access a map with the location of their field service interventions planned for the
day by going to :menuselection:`Planning --> Planning --> My Map`.

.. important::
   For an intervention's location to appear on the map, a valid address must be provided for the
   customer.

Interventions are sorted by their planned date and time in the left pane. Click one to display its
location and details. From there, click :guilabel:`Open` to open the shift form, or
:guilabel:`Navigate to` to get directions from the current location to the intervention's location
in Google Maps.

.. tip::
   - By default, the map displays today's planned interventions. To display all shifts, remove
     the :guilabel:`Start Date: Today` filter in the search bar.
   - Administrators can view a map of all intervention locations for all resources by going to

>      :menuselection:`Planning --> Planning --> Maps`.

Reviewer: As far as I can see, it is like this:

```suggestion
     :menuselection:`Planning --> Maps`.
```
But this view shows all the resources (not just the ones with intervention locations). To only see field service intervention locations, you need to remove the filter.
Or you can access all field service locations (where a valid customer is assigned) via :menu selection: ' Planning --> Planning --> My Map`, and then removing the filters.

Reviewer 2: Let's discuss this (I just added a few points below so I don't forget):

- Should I rephrase to say "a map of all intervention locations, grouped by resource, blabla?
- Because then if you remove the filter, you're not going to see who the intervention is assigned to unless you click it. But maybe then I can just say By default, interventions are grouped by resource blabla, remove the filter to blabla ? I don't know 🙈 
- For the My Map menu, I mentioned it right away on line 295-296 (because from my understanding, this view actually makes the most sense from a usage perspective? and then I'm not sure why technicians would want to see other people's interventions? so yeah I tried to separate : "regular" users' usage and then here in the tip "for admins"

   - You can also display an intervention's location by clicking the shift in the calendar or
     schedule, then clicking :icon:`fa-map-marker` (:guilabel:`View itinerary`).

.. _planning/field-service/itinerary:

Itinerary planning
------------------

To display an itinerary on the map in the Planning app, enable MapBox:

#. Create or sign in to a `MapBox account <https://www.mapbox.com/>`_.

> #. `Create a token <https://docs.mapbox.com/help/getting-started/access-tokens/#adding-url-restrictions-to-access-tokens>`_.

Reviewer: Do you thing we need to specify whether to create a public or secret token?

Reviewer 2: probably, I think I'm just going to create a separate task to dive a bit deeper into this and test it properly again to make sure nothing changed
